src/plugins/WebOfTrust/util/TickerDelayedBackgroundJob.java

```diff
@@ -0,0 +1,363 @@
+            case WAITING:
+                toTERMINATED();
+                return;
+            case RUNNING:
+                toTERMINATING();
+                return;
+        }
+    }
+
+    @Override
+    public synchronized boolean isTerminated() {
+        return state == JobState.TERMINATED;
+    }
+
+    @Override
+    public synchronized void waitForTermination(long timeout) throws InterruptedException {
+        long deadline = System.currentTimeMillis() + timeout;
+        while(timeout > 0 && state != JobState.TERMINATED) {
+            wait(timeout);
+            timeout = deadline - System.currentTimeMillis();
+        }
+    }
+
+    /**
+     * Implementation of {@link #triggerExecution(long)}.
+     * Caller must ensure synchronization on {@code this}.
+     */
+    private void tryEnqueue(long delay) {
+        if (state == JobState.TERMINATING || state == JobState.TERMINATED) {
+            return;
+        }
+        if (delay < 0) {
+            delay = 0;
+        }
+        long newExecutionTime = System.currentTimeMillis() + delay;
+        if (newExecutionTime < nextExecutionTime) {
+            nextExecutionTime = newExecutionTime;
+            if (state == JobState.RUNNING) {
+                // Will automatically schedule this run when the running job finishes.
+                return;
+            }
+            if (state == JobState.WAITING) {
+                // Best-effort attempt at removing the stale job to be replaced.
+                ticker.removeQueuedJob(waitingTickerJob);
+                // Replace the ticker job in case the above fails, so the stale job will not run.
+                waitingTickerJob = createTickerJob();
+            }
+            enqueueWaitingTickerJob(delay);
+        }
+    }
+
+    /**
+     * Schedules the ticker job (which will submit the background job to the executor) at the
+     * ticker with the given delay, or executes the ticker job immediately if the delay is zero.
+     * If the current state is {@code IDLE}, we the state is changed to {@code WAITING} and a new
+     * ticker job is created.
+     * Caller must ensure synchronization on {@code this}.
+     * @param delay the delay in ms
+     */
+    private void enqueueWaitingTickerJob(long delay) {
+        assert(state == JobState.IDLE || state == JobState.WAITING) :
+                "enqueueing ticker job in non-IDLE and non-WAITING state";
+        // Use a unique job for each (re)scheduling to avoid running twice.
+        if (state == JobState.IDLE) {
+            toWAITING();
+        }
+        if (delay > 0) {
+            ticker.queueTimedJob(waitingTickerJob, name + " (waiting)", delay, true, false);
+        } else {
+            waitingTickerJob.run();
+        }
+    }
+
+    /**
+     * Creates a tiny job that instructs the executor to run the background job immediately, to
+     * be executed by the ticker.
+     */
+    private Runnable createTickerJob() {
+        return new FastRunnable() {
```

Comment:
Why is this a FastRunnable? To ensure that PrioritizedTicker will run it on the main thread instead of on the Executor? 
Might want to document that if so.